### ISSUE TITLE
source_catalog unit tests fail with dev photutils

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#6220]
_Issue [JP-2203](https://jira.stsci.edu/browse/JP-2203) was created on JIRA by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies):_

In `jwst` CI on the master branch with dev dependencies, including `photutils` we are seeing 2 `source_catalog` unit tests fail.

<https://github.com/spacetelescope/jwst/runs/3079494003?check_suite_focus=true#step:8:1483>

```
$ pip install -r requirements-dev.txt
$ pytest jwst/source_catalog
============================== test session starts ==============================
platform darwin -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
crds_context: jwst_0738.pmap
rootdir: /Users/jdavies/dev/jwst, configfile: setup.cfg
plugins: cov-2.12.1, doctestplus-0.10.0, requests-mock-1.9.3, asdf-2.9.0.dev6+g887c5be, ci-watson-0.5, openfiles-0.5.0, jwst-1.2.4.dev20+g8a8f148d
collected 3 items                                                               

jwst/source_catalog/tests/test_source_catalog.py FF.                      [100%]

=================================== FAILURES ====================================
___________________________ test_source_catalog[5-2] ____________________________

self = <jwst.source_catalog.source_catalog.JWSTSourceCatalog object at 0x7ff888da2b20>

    def set_segment_properties(self):
        """
        Calculate the segment-based source photometry and morphologies.
    
        The values are set as dynamic attributes.
        """
        segm_cat = SourceCatalog(self.model.data, self.segment_img,
                                 error=self.model.err, kernel=self.kernel,
                                 wcs=self.wcs)
    
        self._xpeak = segm_cat.maxval_xindex
        self._ypeak = segm_cat.maxval_yindex
    
        # rename some columns in the output catalog
        prop_names = {}
        prop_names['isophotal_flux'] = 'segment_flux'
        prop_names['isophotal_flux_err'] = 'segment_fluxerr'
        prop_names['isophotal_area'] = 'area'
    
        for column in self.segment_colnames:
            # define the property name
            prop_name = prop_names.get(column, column)
            try:
>               value = getattr(segm_cat, prop_name)
E               AttributeError: 'SourceCatalog' object has no attribute 'sky_orientation'

jwst/source_catalog/source_catalog.py:631: AttributeError

During handling of the above exception, another exception occurred:

nircam_model = <ImageModel(101, 101)>, npixels = 5, nsources = 2

    @pytest.mark.parametrize('npixels, nsources', ((5, 2), (1000, 1), (5000, 0)))
    def test_source_catalog(nircam_model, npixels, nsources):
    
        step = SourceCatalogStep(snr_threshold=0.5, npixels=npixels,
                                 bkg_boxsize=50, kernel_fwhm=2.0,
                                 save_results=False)
>       cat = step.run(nircam_model)

jwst/source_catalog/tests/test_source_catalog.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniconda3/envs/test/lib/python3.9/site-packages/stpipe/step.py:407: in run
    step_result = self.process(*args)
jwst/source_catalog/source_catalog_step.py:109: in process
    catalog = catobj.catalog
../../miniconda3/envs/test/lib/python3.9/site-packages/astropy/utils/decorators.py:767: in __get__
    val = self.fget(obj)
jwst/source_catalog/source_catalog.py:1300: in catalog
    self.set_segment_properties()
jwst/source_catalog/source_catalog.py:633: in set_segment_properties
    value = getattr(self, prop_name)
../../miniconda3/envs/test/lib/python3.9/site-packages/astropy/utils/decorators.py:767: in __get__
    val = self.fget(obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jwst.source_catalog.source_catalog.JWSTSourceCatalog object at 0x7ff888da2b20>

    @lazyproperty
    def sky_orientation(self):
        """
        The orientation of the source major axis as the position angle
        in degrees measured East of North.
        """
        # NOTE: crpix1 and crpix2 are 1-based values
        skycoord = self.wcs.pixel_to_world(self.model.meta.wcsinfo.crpix1 - 1,
                                           self.model.meta.wcsinfo.crpix2 - 1)
>       _, angle = _pixel_scale_angle_at_skycoord(skycoord, self.wcs)
E       ValueError: too many values to unpack (expected 2)

jwst/source_catalog/source_catalog.py:691: ValueError
__________________________ test_source_catalog[1000-1] __________________________

self = <jwst.source_catalog.source_catalog.JWSTSourceCatalog object at 0x7ff89891c9d0>

    def set_segment_properties(self):
        """
        Calculate the segment-based source photometry and morphologies.
    
        The values are set as dynamic attributes.
        """
        segm_cat = SourceCatalog(self.model.data, self.segment_img,
                                 error=self.model.err, kernel=self.kernel,
                                 wcs=self.wcs)
    
        self._xpeak = segm_cat.maxval_xindex
        self._ypeak = segm_cat.maxval_yindex
    
        # rename some columns in the output catalog
        prop_names = {}
        prop_names['isophotal_flux'] = 'segment_flux'
        prop_names['isophotal_flux_err'] = 'segment_fluxerr'
        prop_names['isophotal_area'] = 'area'
    
        for column in self.segment_colnames:
            # define the property name
            prop_name = prop_names.get(column, column)
            try:
>               value = getattr(segm_cat, prop_name)
E               AttributeError: 'SourceCatalog' object has no attribute 'sky_orientation'

jwst/source_catalog/source_catalog.py:631: AttributeError

During handling of the above exception, another exception occurred:

nircam_model = <ImageModel(101, 101)>, npixels = 1000, nsources = 1

    @pytest.mark.parametrize('npixels, nsources', ((5, 2), (1000, 1), (5000, 0)))
    def test_source_catalog(nircam_model, npixels, nsources):
    
        step = SourceCatalogStep(snr_threshold=0.5, npixels=npixels,
                                 bkg_boxsize=50, kernel_fwhm=2.0,
                                 save_results=False)
>       cat = step.run(nircam_model)

jwst/source_catalog/tests/test_source_catalog.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniconda3/envs/test/lib/python3.9/site-packages/stpipe/step.py:407: in run
    step_result = self.process(*args)
jwst/source_catalog/source_catalog_step.py:109: in process
    catalog = catobj.catalog
../../miniconda3/envs/test/lib/python3.9/site-packages/astropy/utils/decorators.py:767: in __get__
    val = self.fget(obj)
jwst/source_catalog/source_catalog.py:1300: in catalog
    self.set_segment_properties()
jwst/source_catalog/source_catalog.py:633: in set_segment_properties
    value = getattr(self, prop_name)
../../miniconda3/envs/test/lib/python3.9/site-packages/astropy/utils/decorators.py:767: in __get__
    val = self.fget(obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jwst.source_catalog.source_catalog.JWSTSourceCatalog object at 0x7ff89891c9d0>

    @lazyproperty
    def sky_orientation(self):
        """
        The orientation of the source major axis as the position angle
        in degrees measured East of North.
        """
        # NOTE: crpix1 and crpix2 are 1-based values
        skycoord = self.wcs.pixel_to_world(self.model.meta.wcsinfo.crpix1 - 1,
                                           self.model.meta.wcsinfo.crpix2 - 1)
>       _, angle = _pixel_scale_angle_at_skycoord(skycoord, self.wcs)
E       ValueError: too many values to unpack (expected 2)

jwst/source_catalog/source_catalog.py:691: ValueError
============================ short test summary info ============================
FAILED jwst/source_catalog/tests/test_source_catalog.py::test_source_catalog[5-2]
FAILED jwst/source_catalog/tests/test_source_catalog.py::test_source_catalog[1000-1]
========================== 2 failed, 1 passed in 3.27s ==========================
```

So updates to either the code or test code will be needed.

**Comment by larrybradley** [comment]
Fixed by #6206.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2203?focusedCommentId=540156#comment-540156):_

Fixed in #6206
